packages/ui/src/components/va-avatar/VaAvatar.vue

```diff
@@ -26,48 +27,86 @@
 </template>
 
 <script lang="ts">
-import { defineComponent, computed } from 'vue'
+import { defineComponent, ref, watch, computed } from 'vue'
 
-import { useComponentPresetProp, useColors, useTextColor, useSize, useSizeProps, useLoadingProps } from '../../composables'
+import {
+  useSize,
+  useColors,
+  useTextColor,
+  useSizeProps,
+  useLoadingProps,
+  useComponentPresetProp,
+} from '../../composables'
 
-import { VaIcon } from '../va-icon'
-import { VaProgressCircle } from '../va-progress-circle'
+import { VaIcon, VaProgressCircle } from '../index'
+import { useAvatarProps } from './hooks/useAvatarProps'
 
 export default defineComponent({
   name: 'VaAvatar',
+
   components: { VaIcon, VaProgressCircle },
+
   props: {
     ...useLoadingProps,
     ...useSizeProps,
     ...useComponentPresetProp,
-    color: { type: String, default: 'info' },
-    textColor: { type: String },
-    square: { type: Boolean, default: false },
-    icon: { type: String, default: '' },
+    ...useAvatarProps,
+
     src: { type: String, default: null },
+    icon: { type: String, default: '' },
     alt: { type: String, default: '' },
-    fontSize: { type: String, default: '' },
   },
-  setup (props) {
+
+  emits: ['error', 'fallback'],
+
+  setup (props, { emit }) {
     const { getColor } = useColors()
     const colorComputed = computed(() => getColor(props.color))
+    const backgroundColorComputed = computed(() => props.loading || props.src ? 'transparent' : colorComputed.value)
     const { sizeComputed, fontSizeComputed } = useSize(props, 'VaAvatar')
     const { textColorComputed } = useTextColor()
 
     const computedStyle = computed(() => ({
-      color: textColorComputed.value,
-      backgroundColor: props.loading ? 'transparent' : colorComputed.value,
       borderRadius: props.square ? 0 : '',
       fontSize: props.fontSize || fontSizeComputed.value,
-      width: sizeComputed.value,
-      minWidth: sizeComputed.value, // We only define width because common use case would be flex row, for column we expect user to set appropriate styling externally.
-      height: sizeComputed.value,
+    }))
+
+    const hasLoadError = ref(false)
+
+    const srcComputed = computed(() => {
+      if (props.src && props.fallbackSrc && hasLoadError.value) {
+        emit('fallback')
+
+        return props.fallbackSrc
+      }
+
+      return props.src
+    })
+
+    const onLoadError = (event: Event) => {
+      hasLoadError.value = true
+      emit('error', event)
+    }
+
+    watch(() => props.src, () => {
```

Comment:
Mb?

```suggestion
    watch(srcComputed, () => {
```

Reply:
It was on purpose. Here I'm watching for `src` changes, not `src` and `fallback` props together.